addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1047 @@
+        for ev in all_events:
+            rset1._exdate.append(todate(ev.recurrent_id_date))
+        return [d.astimezone(pytz.UTC) for d in rset1]
+
+    def _get_recurrency_end_date(self):
+        if not self.recurrency:
+            return False
+        if self.end_type == 'count' and self.count and self.rrule_type and self.stop:
+            count = self.count + 1
+            delay, mult = {
+                'daily': ('days', 1),
+                'weekly': ('days', 7),
+                'monthly': ('months', 1),
+                'yearly': ('years', 1),
+            }[self.rrule_type]
+
+            deadline = fields.Datetime.from_string(self.stop)
+            return deadline + relativedelta(**{delay: count * mult})
+        return self.final_date
+
+    def _find_my_attendee(self):
+        """
+            Return the first attendee where the user connected has been invited from all the meeting_ids in parameters
+        """
+        for event in self:
+            for attendee in event.attendee_ids:
+                if self.env.user.partner_id.id == attendee.partner_id.id:
+                    return attendee
+        return False
+
+    def get_date_formats(self):
+        lang = self.env.lang
+        lang_params = {}
+        if lang:
+            res_lang = self.env['res.lang'].search([("code", "=", lang)])
+            if res_lang:
+                lang_params = {
+                    'date_format': res_lang.date_format,
+                    'time_format': res_lang.time_format}
+
+        # formats will be used for str{f,p}time() which do not support unicode in Python 2, coerce to str
+        format_date = lang_params.get("date_format", '%B-%d-%Y').encode('utf-8')
+        format_time = lang_params.get("time_format", '%I-%M %p').encode('utf-8')
+        return (format_date, format_time)
+
+    def get_display_time_tz(self, tz=False):
+        context = self.env.context.copy()
+        if tz:
+            context["tz"] = tz
+        return self.with_context(context)._get_display_time(self.start, self.stop, self.duration, self.allday)
+
+    @api.model
+    def _get_display_time(self, start, stop, zduration, zallday):
+        """
+            Return date and time (from to from) based on duration with timezone in string :
+            eg.
+            1) if user add duration for 2 hours, return : August-23-2013 at (04-30 To 06-30) (Europe/Brussels)
+            2) if event all day ,return : AllDay, July-31-2013
+        """
+        tz = self.env.context.get('tz')
+        if not tz:  # tz can have a value False, so dont do it in the default value of get !
+            tz = self.env.context.get(self.env.user.tz)
+        tz = tools.ustr(tz).encode('utf-8')  # make safe for str{p,f}time()
+
+        format_date, format_time = self.with_context(tz=tz).get_date_formats()
+        date = fields.Datetime.context_timestamp(self.with_context(tz=self.env.user.tz), fields.Datetime.from_string(start))
```

Comment:
Is it required to pass self.with_context..., why not only fields.Datetime.context_timestamp(self, ...)